pkg/commands/git_commands/commit_loader.go

```diff
@@ -392,7 +392,16 @@ func (self *CommitLoader) getMergeBase(refName string) string {
 func (self *CommitLoader) getExistingMainBranches() []string {
 	return lo.FilterMap(self.UserConfig.Git.MainBranches,
 		func(branchName string, _ int) (string, bool) {
-			ref := "refs/heads/" + branchName
+			// Try to determine upstream of local main branch
+			if ref, err := self.cmd.New(
+				NewGitCmd("rev-parse").Arg("--symbolic-full-name", branchName+"@{u}").ToArgv(),
+			).DontLog().RunWithOutput(); err == nil {
+				return strings.TrimSpace(ref), true
+			}
+
+			// If this failed, a local branch for this main branch doesn't exist or it
+			// has no upstream configured. Try looking for one in the "origin" remote.
+			ref := "refs/remotes/origin/" + branchName
```

Comment:
What are your thoughts on finally falling back to the local branch itself, assuming there's no upstream and there's no branch in origin? I'm thinking of a use case where somebody is locally using git for some project but they're not pushing it anywhere

Reply:
Wanna have another look @jesseduffield? This is ready to go, and I'm missing the improved behavior when I work on master. :)